liberty/Liberty.cc

```diff
@@ -1989,6 +1999,18 @@ LibertyCell::ensureVoltageWaveforms(const DcalcAnalysisPtSeq &dcalc_aps)
   }
 }
 
+void
+LibertyCell::setFootprintIndex(LibertyCellFootprintIndex footprint_index)
+{
+  footprint_index_ = footprint_index;
+}
+
+const char*
+LibertyCell::footprint() const
+{
+  return libertyLibrary()->footprints_[footprint_index_];
```

Comment:
footprint is an optional field so you need to handle the case where it is unset in Liberty.